src/web/extractors.rs

```diff
@@ -1353,6 +1361,20 @@ impl FromRequest for BsoQueryParams {
                             .into());
                         }
                     }
+                } else {
```

Comment:
Following my comment below -- so instead of the `offset.offset == 0` check I think this is what we want:

```suggestion
                } else if params.sort != Sorting::Index {
```

no timestamp + non-index. I still think a couple extra tests would be helpful for this.

But argh, this leads to another issue I hadn't considered: mysql doesn't implement an `encode_offset` equiv. yet. We either can't enforce this strictness until that's implemented or it would have to be Spanner only.

I think we forgot to log an issue for bringing this over to mysql, might be a _Good First Issue_.

Reply:
Ok, great. This is much better. Thanks.

Well, should we wait to merge this pr until mysql has the encode_offset code then?

Reply:
Do you think it'd be a good first issue? We could just implement it now too, I suspect it'd be pretty straightforward w Spanner already done (would probably use the same `encode_offset` so maybe just a matter of plugging it in)

Reply:
I'll just do it in this pr and get it over with